tests/src/Functional/DeveloperAppUITest.php

```diff
@@ -593,4 +594,27 @@ public function testBreadcrumbOnDeveloperAppPages() {
     $this->assertEquals(Url::fromRoute('entity.developer_app.collection_by_developer', ['user' => $user->id()])->toString(), $breadcrumb_links[2]->getAttribute('href'));
   }
 
+  /**
+   * Validates access on the user/[uid]/create-app route.
+   *
+   * TODO Move this to \Drupal\Tests\apigee_edge\Functional\DeveloperAppPermissionTest
```

Comment:
"Moving" means, refactor the mentioned test to validate this properly, because it should have validated this.